src/category_theory/limits/shapes/pullbacks.lean

```diff
@@ -661,6 +688,336 @@ by { ext; simp [← G.map_comp] }
+lemma has_pushout_symmetry [has_pushout f g] : has_pushout g f :=
+⟨⟨⟨pushout_cocone.mk _ _ pushout.condition.symm,
+  pushout_cocone.flip_is_colimit (pushout_is_pushout _ _)⟩⟩⟩
+
+local attribute [instance] has_pushout_symmetry
+
+/-- The isomorphism `Y ⨿[X] Z ≅ Z ⨿[X] Y`. -/
+def pushout_symmetry [has_pushout f g] :
+  pushout f g ≅ pushout g f :=
+is_colimit.cocone_point_unique_up_to_iso
+  (pushout_cocone.flip_is_colimit (pushout_is_pushout f g) :
+    is_colimit (pushout_cocone.mk _ _ pushout.condition.symm))
+  (colimit.is_colimit _)
+
+@[simp, reassoc] lemma inl_comp_pushout_symmetry_hom [has_pushout f g] :
+  pushout.inl ≫ (pushout_symmetry f g).hom = pushout.inr :=
+(colimit.is_colimit (span f g)).comp_cocone_point_unique_up_to_iso_hom
+  (pushout_cocone.flip_is_colimit (pushout_is_pushout g f)) _
+
+@[simp, reassoc] lemma inr_comp_pushout_symmetry_hom [has_pushout f g] :
+  pushout.inr ≫ (pushout_symmetry f g).hom = pushout.inl :=
+(colimit.is_colimit (span f g)).comp_cocone_point_unique_up_to_iso_hom
+  (pushout_cocone.flip_is_colimit (pushout_is_pushout g f)) _
+
+@[simp, reassoc] lemma inl_comp_pushout_symmetry_inv [has_pushout f g] :
+  pushout.inl ≫ (pushout_symmetry f g).inv = pushout.inr := by simp [iso.comp_inv_eq]
+
+@[simp, reassoc] lemma inr_comp_pushout_symmetry_inv [has_pushout f g] :
+  pushout.inr ≫ (pushout_symmetry f g).inv = pushout.inl := by simp [iso.comp_inv_eq]
+
+end pushout_symmetry
+
+section pullback_left_iso
+
+open walking_cospan
+
+variables (f : X ⟶ Z) (g : Y ⟶ Z) [is_iso f]
+
+/-- If `f : X ⟶ Z` is iso, then `X ×[Z] Y ≅ Y`. This is the explicit limit cone. -/
+def pullback_cone_of_left_iso : pullback_cone f g :=
+pullback_cone.mk (g ≫ inv f) (𝟙 _) $ by simp
+
+@[simp] lemma pullback_cone_of_left_iso_X :
+  (pullback_cone_of_left_iso f g).X = Y := rfl
+
+@[simp] lemma pullback_cone_of_left_iso_fst :
+  (pullback_cone_of_left_iso f g).fst = g ≫ inv f := rfl
+
+@[simp] lemma pullback_cone_of_left_iso_snd :
+  (pullback_cone_of_left_iso f g).snd = 𝟙 _ := rfl
+
+@[simp] lemma pullback_cone_of_left_iso_π_app_none :
+  (pullback_cone_of_left_iso f g).π.app none = g := by { delta pullback_cone_of_left_iso, simp }
+
+@[simp] lemma pullback_cone_of_left_iso_π_app_left :
+  (pullback_cone_of_left_iso f g).π.app left = g ≫ inv f := rfl
+
+@[simp] lemma pullback_cone_of_left_iso_π_app_right :
+  (pullback_cone_of_left_iso f g).π.app right = 𝟙 _ := rfl
+
+/-- Verify that the constructed limit cone is indeed a limit. -/
+def pullback_cone_of_left_iso_is_limit :
+  is_limit (pullback_cone_of_left_iso f g) :=
+pullback_cone.is_limit_aux' _ (λ s, ⟨s.snd, by simp [← s.condition_assoc]⟩)
+
+lemma has_pullback_of_left_iso : has_pullback f g :=
+⟨⟨⟨_, pullback_cone_of_left_iso_is_limit f g⟩⟩⟩
+
+local attribute [instance] has_pullback_of_left_iso
+
+instance pullback_snd_iso_of_left_iso : is_iso (pullback.snd : pullback f g ⟶ _) :=
+begin
+  constructor,
+  refine ⟨pullback.lift (g ≫ inv f) (𝟙 _) (by simp), _, by simp⟩,
```

Comment:
This shortens to
```suggestion
  refine ⟨⟨pullback.lift (g ≫ inv f) (𝟙 _) (by simp), _, by simp⟩⟩,
```

Reply:
Of course, the same comment applies to the other five places with similar proofs.

Reply:
I forgot about them. Thanks!